src/components/ShareMyLocationButton.js

```diff
@@ -28,20 +28,33 @@ const ShareMyLocationButton = ({ disabled = false }) => {
 
   const handleShareLiveLocation = () => {
     if (navigator.geolocation) {
-      if (trackingId === null) {
-        const newTrackingId = navigator.geolocation.watchPosition(
-          handleSuccess,
-          handleError,
-        );
-        setTrackingId(newTrackingId);
-      }
+      const trackingId = navigator.geolocation.watchPosition(
+        handleSuccess,
+        handleError,
+      );
+
+      setTrackingId(trackingId);
+      setUserLocationShared(true);
     }
   };
 
+  const handleStopSharing = () => {
+    if (userLocationShared && trackingId) {
+      setTrackingId(navigator.geolocation.clearWatch(trackingId));
```

Comment:
`navigator.geolocation.clearWatch` doesn't return any value according to the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/clearWatch) (you're setting `undefined`, in this case, it doesn't cause some problem), but I think it was much clear. First, clear the watch, and next in other line set the tracking value to `null`)

Reply:
yes, exactly that, we are updating state to ``undefined``, it is  completely valid😄